https://github.com/fraunhoferhhi/vvdec

```
In file included from /usr/ports/multimedia/vvdec/work/vvdec-2.2.0/source/Lib/vvdec/vvdec.cpp:43:
/usr/ports/multimedia/vvdec/work/vvdec-2.2.0/source/Lib/vvdec/./vvdecimpl.h:228:14: error: cast from pointer to smaller type 'TRet' (aka 'int') loses information
      return (TRet) NULL;
             ^~~~~~~~~~~
/usr/ports/multimedia/vvdec/work/vvdec-2.2.0/source/Lib/vvdec/vvdec.cpp:205:21: note: in instantiation of function template specialization 'vvdec::VVDecImpl::catchExceptions<int (vvdec::VVDecImpl::*)(const vvdecParams &, void *(*)(void *, vvdecComponentType, unsigned int, unsigned int, void **), void (*)(void *, void *)), vvdecParams, std::nullptr_t, std::nullptr_t>' requested here
  int ret = decCtx->catchExceptions( &vvdec::VVDecImpl::init, *params, nullptr, nullptr );
                    ^
In file included from /usr/ports/multimedia/vvdec/work/vvdec-2.2.0/source/Lib/vvdec/vvdec.cpp:43:
/usr/ports/multimedia/vvdec/work/vvdec-2.2.0/source/Lib/vvdec/./vvdecimpl.h:228:14: error: cast from pointer to smaller type 'TRet' (aka 'int') loses information
      return (TRet) NULL;
             ^~~~~~~~~~~
/usr/ports/multimedia/vvdec/work/vvdec-2.2.0/source/Lib/vvdec/vvdec.cpp:247:21: note: in instantiation of function template specialization 'vvdec::VVDecImpl::catchExceptions<int (vvdec::VVDecImpl::*)(const vvdecParams &, void *(*)(void *, vvdecComponentType, unsigned int, unsigned int, void **), void (*)(void *, void *)), vvdecParams, void *(*)(void *, vvdecComponentType, unsigned int, unsigned int, void **), void (*)(void *, void *)>' requested here
  int ret = decCtx->catchExceptions( &vvdec::VVDecImpl::init, *params, callbackBufAlloc, callbackBufUnref );
                    ^
In file included from /usr/ports/multimedia/vvdec/work/vvdec-2.2.0/source/Lib/vvdec/vvdec.cpp:43:
/usr/ports/multimedia/vvdec/work/vvdec-2.2.0/source/Lib/vvdec/./vvdecimpl.h:228:14: error: cast from pointer to smaller type 'TRet' (aka 'int') loses information
      return (TRet) NULL;
             ^~~~~~~~~~~
/usr/ports/multimedia/vvdec/work/vvdec-2.2.0/source/Lib/vvdec/vvdec.cpp:271:22: note: in instantiation of function template specialization 'vvdec::VVDecImpl::catchExceptions<int (vvdec::VVDecImpl::*)()>' requested here
  const int ret = d->catchExceptions( &vvdec::VVDecImpl::uninit );
                     ^
In file included from /usr/ports/multimedia/vvdec/work/vvdec-2.2.0/source/Lib/vvdec/vvdec.cpp:43:
/usr/ports/multimedia/vvdec/work/vvdec-2.2.0/source/Lib/vvdec/./vvdecimpl.h:228:14: error: cast from pointer to smaller type 'TRet' (aka 'int') loses information
      return (TRet) NULL;
             ^~~~~~~~~~~
/usr/ports/multimedia/vvdec/work/vvdec-2.2.0/source/Lib/vvdec/vvdec.cpp:306:13: note: in instantiation of function template specialization 'vvdec::VVDecImpl::catchExceptions<int (vvdec::VVDecImpl::*)(vvdecAccessUnit &, vvdecFrame **), vvdecAccessUnit, vvdecFrame **>' requested here
  return d->catchExceptions( &vvdec::VVDecImpl::decode, *accessUnit, frame );
            ^
In file included from /usr/ports/multimedia/vvdec/work/vvdec-2.2.0/source/Lib/vvdec/vvdec.cpp:43:
/usr/ports/multimedia/vvdec/work/vvdec-2.2.0/source/Lib/vvdec/./vvdecimpl.h:228:14: error: cast from pointer to smaller type 'TRet' (aka 'int') loses information
      return (TRet) NULL;
             ^~~~~~~~~~~
/usr/ports/multimedia/vvdec/work/vvdec-2.2.0/source/Lib/vvdec/vvdec.cpp:320:13: note: in instantiation of function template specialization 'vvdec::VVDecImpl::catchExceptions<int (vvdec::VVDecImpl::*)(vvdecFrame **), vvdecFrame **>' requested here
  return d->catchExceptions( &vvdec::VVDecImpl::flush, frame );
            ^
In file included from /usr/ports/multimedia/vvdec/work/vvdec-2.2.0/source/Lib/vvdec/vvdec.cpp:43:
/usr/ports/multimedia/vvdec/work/vvdec-2.2.0/source/Lib/vvdec/./vvdecimpl.h:228:14: error: cast from pointer to smaller type 'TRet' (aka 'int') loses information
      return (TRet) NULL;
             ^~~~~~~~~~~
```

NULL isn't generally defined as zero, instead it is defined as nullptr.


You need to use "0" in these places.

Version: 2.2.0
clang-16
FreeBSD 14.0

--- a/source/Lib/vvdec/vvdecimpl.h
+++ b/source/Lib/vvdec/vvdecimpl.h
@@ -220,12 +220,13 @@ private:
 template<class MembFunc, class... Args>
 inline auto VVDecImpl::catchExceptions( MembFunc fn, Args... args )
 {
-  using TRet = decltype( ( this->*fn )( args... ) );
-  // helper to either return an error value or nullptr based on the wrapped function's return type
+  using TRet = decltype( ( this->*fn )( args... ) ); // return type of the wapped function
+
+  // helper function to either return an error value or a null-pointer based on the wrapped function's return type
   static auto returnErrOrNullptr = []( intptr_t err )
   {
     if( std::is_pointer<TRet>() )
-      return (TRet) NULL;
+      return (TRet) 0;
     return (TRet) err;
   };
 

